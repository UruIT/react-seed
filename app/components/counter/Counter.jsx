import React, { Component } from 'react';
import styles from './Counter.scss';

const increment = step => ({ value }) => ({ value: value + step });

class Counter extends Component {
	constructor(props) {
		super(props);
		this.state = {
			value: 0
		};
	}
	handleIncClick = e => {
		e.preventDefault();
		this.setState(increment(1));
	}
	handleDecClick = e => {
		e.preventDefault();
		this.setState(increment(-1));
	}
	render() {
		return (
			<div className={styles.container} >
				<div>{ this.state.value }</div>
				<button onClick={this.handleDecClick}>&#x2212;</button>
				<button onClick={this.handleIncClick}>&#x2b;</button>
			</div>
		);
	}
}

export default Counter;
