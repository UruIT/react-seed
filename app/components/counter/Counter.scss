@import '../../styles/box-shadow.scss';

.container {
	background: #fbfb09;
	@include box-shadow(1px, 1px, 4px,#4c4a4a);
	border: 2px solid #222;
	width: 15rem;
	margin: auto;

	div {
		font-size: 7rem;
	}

	button {
		display: inline-block;
		font-weight: bold;
		font-size: 3rem;
		cursor: pointer;
		border: 2px solid #222;
		background:inherit;
		padding: 0.2rem;
		margin: 1rem;
		width: 3.5rem;
	}
}
