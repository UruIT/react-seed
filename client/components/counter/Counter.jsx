import React, { Component } from 'react';
import styles from './counter.scss';

const leap = step => ({ value }) => ({ value: value + step });

class Counter extends Component {
	constructor(props) {
		super(props);

		this.handleIncrement = this.handleIncrement.bind(this);
		this.handleDecrement = this.handleDecrement.bind(this);
		this.state = { value: 0 };
	}

	handleIncrement() {
		this.setState(leap(1));
	}

	handleDecrement() {
		this.setState(leap(-1));
	}

	render() {
		return (
			<div className={styles.container}>
				<div className={styles.value}>{this.state.value}</div>
				<div className={styles.controls}>
					<a onClick={this.handleDecrement}>âˆ’</a>
					<a onClick={this.handleIncrement}>+</a>
				</div>
			</div>
		);
	}
}

export default Counter;
