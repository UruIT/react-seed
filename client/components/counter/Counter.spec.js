import React from 'react';
import Counter from './Counter';
import ShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';
import withContext from '../__mocks__/context-hoc';

const CONTEXT = {
	store: {
		getState: jest.fn(() => ({ counter: 0 })),
		subscribe: jest.fn(),
		dispatch: jest.fn()
	}
};

describe('<Counter/>', () => {
	const CounterWrapper = withContext(CONTEXT)(Counter);

	it('match <Connect(Counter)/>', () => {
		const renderer = new ShallowRenderer();

		const tree = renderer.render(<CounterWrapper />);

		expect(tree).toMatchSnapshot();
	});

	it('render primitives', () => {
		const tree = renderer.create(<CounterWrapper />).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
