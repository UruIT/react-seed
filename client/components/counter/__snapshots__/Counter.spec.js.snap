// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> 1`] = `
<div
  className="container"
>
  <div />
  <a
    onClick={[Function]}
  >
    âˆ’
  </a>
  <a
    onClick={[Function]}
  >
    +
  </a>
</div>
`;
